pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/CommentDefaultAccessModifierRule.java

```diff
@@ -89,10 +91,16 @@ public Object visit(final ASTConstructorDeclaration decl, Object data) {
     }
 
     private boolean shouldReport(final AbstractJavaAccessNode decl) {
-        List<ASTClassOrInterfaceDeclaration> parentClassOrInterface = decl
-                .getParentsOfType(ASTClassOrInterfaceDeclaration.class);
-        // ignore if is a Interface
-        return !parentClassOrInterface.isEmpty() && !parentClassOrInterface.get(0).isInterface()
+        final AbstractAnyTypeDeclaration parentClassOrInterface = decl
+                .getFirstParentOfType(AbstractAnyTypeDeclaration.class);
+        
+        final boolean isConcreteClass = parentClassOrInterface instanceof ASTClassOrInterfaceDeclaration
+                && !((ASTClassOrInterfaceDeclaration) parentClassOrInterface).isInterface();
+        final boolean isEnumConstructor = parentClassOrInterface instanceof ASTEnumDeclaration
+                && decl instanceof ASTConstructorDeclaration;
```

Comment:
I think, you can use here `net.sourceforge.pmd.lang.java.ast.ASTAnyTypeDeclaration.getTypeKind()` to figure out, whether it's an enum, interface or class.

```
boolean isConcreteClass = parentClassOrInterface.getTypeKind() == ASTAnyTypeDeclaration.TypeKind.CLASS;
boolean isEnumConstructor = parentClassOrInterface.getTypeKind() == ASTAnyTypeDeclaration.TypeKind.ENUM && decl instanceof ASTConstructor;
```

Hm... I hoped, it would pan out nicer... but at least, it leverages the ASTAnyTypeDeclaration API :smile:

Reply:
Feel free to do so!